libpod/container_internal_linux.go

```diff
@@ -205,6 +205,27 @@ func (c *Container) resolveWorkDir() error {
 		// the path exists on the container.
 		if err != nil {
 			if os.IsNotExist(err) {
+
+				// We cannot create workdir since explicit --workdir is set in config but workdir could also be a symlink.
+				// If its a symlink lets check if resolved link is present on the container or not, If we can resolve symlink
+				// and resolved link is present on the container then return nil cause its a vaild use-case.
+				workDirInfo, err := os.Lstat(resolvedWorkdir)
+				if workDirInfo != nil && err == nil && (workDirInfo.Mode()&os.ModeSymlink == os.ModeSymlink) {
+					resolvedSymlink, err := os.Readlink(resolvedWorkdir)
+					if err == nil && resolvedSymlink != "" {
+						_, resolvedSymlinkWorkdir, err := c.resolvePath(c.state.Mountpoint, resolvedSymlink)
+						if err == nil && resolvedSymlinkWorkdir != "" {
+							_, err := os.Stat(resolvedSymlinkWorkdir)
```

Comment:
I think it is better to remove this check.  The resolved path could be on a volume.

Reply:
@giuseppe I have added a check just before this line to check if resolved link exists on a `volume` or external `mount` and then doing a `stat` for inside container check. Does that addresses this ?

The reason I'm doing a stat for inside container check is if resolved `link` does not exists inside container then `run` would fail later in the code.